components/server/src/server.ts

```diff
@@ -306,6 +308,7 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
             res.send(this.config.version);
         });
         app.use(this.oauthController.oauthRouter);
+        app.use("/feature-flags", this.featureFlagController.apiRouter);
```

Comment:
Will caddy forward the full request (including credentials) to this endpoint to authenticate it?

What does currently authenticate this endpoint? Inside this method, there's both `/live` (which is unathenticated) and also other endpoints which do require auth.

Reply:
> Will caddy forward the full request (including credentials) to this endpoint to authenticate it?

I assume so, Caddy is doing this in the preview environment and so I assume it will behave the same in production.

> What does currently authenticate this endpoint? Inside this method, there's both /live (which is unathenticated) and also other endpoints which do require auth.

Authentication is handled by passport.js, which gives us the `user` on the request. This applies to all handlers defined here - the liveness one doesn't check for authentication.

Reply:
We've used sub-requests in Caddy for extension upload in the past. That should work as anticipated.
If that's helpful: https://caddyserver.com/docs/caddyfile/directives/forward_auth#tailscale

Reply:
Thanks, I plan to use this endpoint as part of a Caddy subrequest 👍